backend/api/communication/src/main/java/co/airy/core/api/communication/SendMessageController.java

```diff
@@ -78,22 +78,33 @@
         }
 
         final String userId = principalAccess.getUserId(auth);
+        final String messageId = UUID.randomUUID().toString();
+        final String channelId = Optional.ofNullable(channel).map((c) -> c.getId()).orElse("");
 
         final Message message = Message.newBuilder()
-                .setId(UUID.randomUUID().toString())
-                .setChannelId(channel.getId())
+                .setId(messageId)
+                .setChannelId(channelId)
                 .setSourceRecipientId(payload.getSourceRecipientId())
                 .setContent(objectMapper.writeValueAsString(payload.getMessage()))
-                .setConversationId(conversationId)
+                // If channelId is not set we should not set the conversationId either
+                .setConversationId(channelId != "" ? conversationId : "")
                 .setHeaders(Map.of())
                 .setDeliveryState(DeliveryState.PENDING)
-                .setSource(channel.getSource())
+                .setSource(Optional.ofNullable(channel).map((c) -> c.getSource()).orElse(""))
                 .setSenderId(userId)
                 .setSentAt(Instant.now().toEpochMilli())
                 .setIsFromContact(false)
                 .build();
```

Comment:
@chrismatix I am not sure if the system can tolerate empty values on the `ConversationId` `ChannelId` and `Source`

Reply:
It can't actually. So my suggestion would be, having thought about this again, that we don't even need to store the placeholder message. We can only store the error metadata. 

This way the sender gets informed async without creating broken conversations.